### PR TITLE
cli: add policy list and info to new scaling cmd.

### PR DESCRIPTION
**Pull request by jrasell**
This adds the ability to detail scaling policies via the CLI.

Relates #8649
Docs to be added in future PR.

### CODE REVIEW

**Review by cgbaker — APPROVED** [review]
LGTM

File: command/scaling_policy_info.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+// Synopsis satisfies the cli.Command Synopsis function.
+func (s *ScalingPolicyInfoCommand) Synopsis() string {
+	return "Display an individual Nomad scaling policy"
+}
+
+func (s *ScalingPolicyInfoCommand) AutocompleteFlags() complete.Flags {
+	return mergeAutocompleteFlags(s.Meta.AutocompleteFlags(FlagSetClient),
+		complete.Flags{
+			"-json": complete.PredictNothing,
+			"-t":    complete.PredictAnything,
```
Comment (lgfa29) [inline command/scaling_policy_info.go:50]: Oh cool! Is this this the first command to support this?

File: command/scaling_policy_list.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+	}
+
+	// Get the HTTP client.
+	client, err := s.Meta.Client()
+	if err != nil {
+		s.Ui.Error(fmt.Sprintf("Error initializing client: %s", err))
+		return 1
+	}
+
+	policies, _, err := client.Scaling().ListPolicies(nil)
```
Comment (lgfa29) [inline command/scaling_policy_list.go:84]: I think we need a `-namespace` argument for listing policies?

File: command/scaling_policy_info.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+	policyID := args[0]
+
+	// Get the HTTP client.
+	client, err := s.Meta.Client()
+	if err != nil {
+		s.Ui.Error(fmt.Sprintf("Error initializing client: %s", err))
+		return 1
+	}
+
+	policy, _, err := client.Scaling().GetPolicy(policyID, nil)
```
Comment (lgfa29) [inline command/scaling_policy_info.go:86]: We should probably add something like `GetPolicyByPrefixID` so this command can be used with partial ID prefix matching.

**Review by lgfa29 — APPROVED** [review]
Looking good!

One thing I noticed is that the commands don't support Namespaces. We can add that in a follow-up PR though.

It would be nice to have ID prefix matching as well, but that's more of a change in the `api` than here.

File: command/scaling_policy_info.go (hunk 1 above)
Comment (jrasell) [reply command/scaling_policy_info.go:50]: There are quite a few that already support this; although i'll admit I have never used them.

File: command/scaling_policy_list.go (hunk 2 above)
Comment (jrasell) [reply command/scaling_policy_list.go:84]: Namespace is included in the command meta when using `FlagSetClient`. The client subsequently gets its config correctly populated [here](https://github.com/hashicorp/nomad/blob/97d03887c767c0e9c78a2cdd15825544c2010dbe/command/meta.go#L124).

File: command/scaling_policy_info.go (hunk 3 above)
Comment (jrasell) [reply command/scaling_policy_info.go:86]: Agreed.

File: command/scaling_policy_list.go (hunk 2 above)
Comment (lgfa29) [reply command/scaling_policy_list.go:84]: Magical!

```shellsession
$ nomad scaling policy list -namespace=default
ID                                    Enabled  Target
52cdb238-c37c-662a-5c6e-7af1b5978757  true     Namespace:default,,
aaf6a549-80c4-8716-f70f-b59d5cd0395d  false    Namespace:default,Job:webapp,Group:demo

$ nomad scaling policy list -namespace=dev
ID                                    Enabled  Target
2b31db60-ca35-e992-d223-527e82b850aa  true     Namespace:dev,,
````

Comment (jrasell) [reply command/scaling_policy_list.go:84]: Note: trailing commas are now removed.

**Pull request merged**